File: src/googleSheets.ts
import { google } from 'googleapis'

export async function logRequest(data: unknown): Promise<string> {
  const auth = await google.auth.getClient({
    scopes: ['https://www.googleapis.com/auth/spreadsheets']
  })
  const sheets = google.sheets({ version: 'v4', auth })
  const spreadsheetId = process.env.GOOGLE_SHEET_ID!
  await sheets.spreadsheets.values.append({
    spreadsheetId,
    range: 'Requests!A:Z',
    valueInputOption: 'RAW',
    requestBody: {
      values: [[new Date().toISOString(), JSON.stringify(data)]]
    }
  })
  return 'Request logged to Google Sheets'
}

File: pyagent/google_sheets.py
from googleapiclient.discovery import build
from google.oauth2 import service_account
import os

SCOPES = ['https://www.googleapis.com/auth/spreadsheets']

async def log_request(data: dict) -> str:
    creds = service_account.Credentials.from_service_account_file(
        os.environ['GOOGLE_SERVICE_ACCOUNT'], scopes=SCOPES
    )
    service = build('sheets', 'v4', credentials=creds)
    sheet = service.spreadsheets()
    spreadsheet_id = os.environ['GOOGLE_SHEET_ID']
    body = {'values': [[data]]}
    await sheet.values().append(
        spreadsheetId=spreadsheet_id,
        range='Requests!A:Z',
        valueInputOption='RAW',
        body=body
    ).execute()
    return 'Request logged to Google Sheets'
